src/sentry_logs.c

```diff
@@ -0,0 +1,416 @@
+        break;
+    }
+    case 's': {
+        const char *val = va_arg(*args_copy, const char *);
+        if (val) {
+            sentry_value_set_by_key(
+                param_obj, "value", sentry_value_new_string(val));
+        } else {
+            sentry_value_set_by_key(
+                param_obj, "value", sentry_value_new_string("(null)"));
+        }
+        sentry_value_set_by_key(
+            param_obj, "type", sentry_value_new_string("string"));
+        break;
+    }
+    case 'p': {
+        void *val = va_arg(*args_copy, void *);
+        char ptr_str[32];
+        snprintf(ptr_str, sizeof(ptr_str), "%p", val);
+        sentry_value_set_by_key(
+            param_obj, "value", sentry_value_new_string(ptr_str));
+        sentry_value_set_by_key(
+            param_obj, "type", sentry_value_new_string("string"));
+        break;
+    }
+    default:
+        // Unknown format specifier, skip the argument
+        (void)va_arg(*args_copy, void *);
+        sentry_value_set_by_key(
+            param_obj, "value", sentry_value_new_string("(unknown)"));
+        sentry_value_set_by_key(
+            param_obj, "type", sentry_value_new_string("string"));
+        break;
+    }
+
+    return param_obj;
+}
+
+static const char *
+skip_flags(const char *fmt_ptr)
+{
+    while (*fmt_ptr
+        && (*fmt_ptr == '-' || *fmt_ptr == '+' || *fmt_ptr == ' '
+            || *fmt_ptr == '#' || *fmt_ptr == '0')) {
+        fmt_ptr++;
+    }
+    return fmt_ptr;
+}
+
+static const char *
+skip_width(const char *fmt_ptr)
+{
+    while (*fmt_ptr && (*fmt_ptr >= '0' && *fmt_ptr <= '9')) {
+        fmt_ptr++;
+    }
+    return fmt_ptr;
+}
+
+static const char *
+skip_precision(const char *fmt_ptr)
+{
+
+    if (*fmt_ptr == '.') {
+        fmt_ptr++;
+        while (*fmt_ptr && (*fmt_ptr >= '0' && *fmt_ptr <= '9')) {
+            fmt_ptr++;
+        }
+    }
+    return fmt_ptr;
+}
+
+static void
+populate_message_parameters(
+    sentry_value_t attributes, const char *message, va_list args)
+{
+    if (!message || sentry_value_is_null(attributes)) {
+        return;
+    }
+
+    const char *fmt_ptr = message;
+    int param_index = 0;
+    va_list args_copy;
+    va_copy(args_copy, args);
+
+    while (*fmt_ptr) {
+        // Find the next format specifier
+        if (*fmt_ptr == '%') {
+            fmt_ptr++; // Skip the '%'
+
+            if (*fmt_ptr == '%') {
+                // Escaped '%', not a format specifier
+                fmt_ptr++;
+                continue;
+            }
+
+            fmt_ptr = skip_flags(fmt_ptr);
+            fmt_ptr = skip_width(fmt_ptr);
+            fmt_ptr = skip_precision(fmt_ptr);
+
+            // Skip length modifiers
+            while (*fmt_ptr
+                && (*fmt_ptr == 'h' || *fmt_ptr == 'l' || *fmt_ptr == 'L'
+                    || *fmt_ptr == 'z' || *fmt_ptr == 'j' || *fmt_ptr == 't')) {
+                fmt_ptr++;
+            }
```

Comment:
Since we now always store with the largest value types available, and there is no need to extract length modifiers, this could also be extracted to a `skip_length` function, right?